src/Features/Core/Portable/Completion/CompletionProvider.cs

```diff
@@ -38,6 +38,11 @@ public virtual bool ShouldTriggerCompletion(SourceText text, int caretPosition,
             return false;
         }
 
+        internal virtual async Task<bool> IsSyntacticTriggerCharacterAsync(Document document, int caretPosition, CompletionTrigger trigger, OptionSet options, CancellationToken cancellationToken)
```

Comment:
protected?

Reply:
Needs to be called from the CompletionServiceWithProviders...

Reply:
👍